app/src/main/java/com/itssuryansh/taaveez/Notes.kt

```diff
@@ -196,12 +214,22 @@ class Notes : AppCompatActivity() {
                     ShareNotes(ShareId, NotesDao)
                 },
             )
-
+        }
+        if (NotesList.isNotEmpty()) {
+            //this if-condition is invoked if the list is not empty.
+            /*
+            * Also , this if condition is also invoked for every new update in the notesList.
+            * */
             binding?.rvItemsPoem?.layoutManager = LinearLayoutManager(this)
```

Comment:
dont repeat this step all the time, rather create a function to initialize RecyclerView
Inside it
- Initialize adapter, you wont need to pass List of Notes, when you use `RecyclerView.ListAdapter`
- set up recyclerView